### PR TITLE
feat(scheduled-tasks): add `delete`, `list`, `listRepeated` & `get` methods

### PR DESCRIPTION
**Pull request by FC5570**
Adds 4 new methods:
1. [delete](https://github.com/OptimalBits/bull/blob/HEAD/REFERENCE.md#jobremove): Deletes/removes a job.
2. [list](https://github.com/OptimalBits/bull/blob/HEAD/REFERENCE.md#queuegetjobs): Lists jobs based on the type provided.
3. [listRepeated](https://github.com/OptimalBits/bull/blob/HEAD/REFERENCE.md#queuegetrepeatablejobs): Lists repeatable jobs.
4. [get](https://github.com/OptimalBits/bull/blob/HEAD/REFERENCE.md#queuegetjob): Get's a job of the id provided.

### CODE REVIEW

**Comment by favna** [comment]
This won't actually work because `container.tasks.strategy` has the type of the base thing we implement. Instead these methods should be implemented in the ScheduledTaskHandler which then call the methods in the strategies.

File: packages/scheduled-tasks/src/lib/strategies/ScheduledTaskRedisStrategy.ts [hunk 1]
```diff
@@ -70,6 +70,39 @@ export class ScheduledTaskRedisStrategy implements ScheduledTaskBaseStrategy {
 		}
 	}
 
+	public async delete(id: JobId) {
+		if (!this.bullClient) {
+			return;
+		}
+
+		const job = await this.bullClient.getJob(id);
```
Comment (jurienhamaker) [inline packages/scheduled-tasks/src/lib/strategies/ScheduledTaskRedisStrategy.ts:78]: `this.get(id)`

File: packages/scheduled-tasks/src/lib/strategies/ScheduledTaskRedisStrategy.ts [hunk 2]
```diff
@@ -70,6 +80,41 @@ export class ScheduledTaskRedisStrategy implements ScheduledTaskBaseStrategy {
...
+
+		return this.bullClient.getRepeatableJobs(start, end, asc);
+	}
+
+	public get(id: JobId): undefined | Promise<Job> {
+		if (!this.bullClient) {
+			return;
+		}
+
+		return this.get(id);
```
Comment (jurienhamaker) [inline packages/scheduled-tasks/src/lib/strategies/ScheduledTaskRedisStrategy.ts:115]: This is now recursive and never ending

**Review by favna — APPROVED** [review]

**Comment by favna** [comment]
@all-contributors please add @FC5570 for code

**Pull request merged**